src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/SourceMappingParser.cs

```diff
@@ -42,20 +42,53 @@ public async Task<IReadOnlyCollection<SourceMapping>> ParseMappings(string vmrPa
         };
 
         using var stream = File.Open(mappingFile.FullName, FileMode.Open);
-        var mappings = await JsonSerializer.DeserializeAsync<SourceMappingSetting[]>(stream, options)
+        var settings = await JsonSerializer.DeserializeAsync<SourceMappingFile>(stream, options)
             ?? throw new Exception($"Failed to deserialize {SourceMappingsFileName}");
 
-        return mappings
-            .Select(m => new SourceMapping(
-                Name: m.Name ?? throw new InvalidOperationException($"Missing `name` in {SourceMappingsFileName}"),
-                Version: m.Version,
-                DefaultRemote: m.DefaultRemote ?? throw new InvalidOperationException($"Missing `defaultRemote` in {SourceMappingsFileName}"),
-                DefaultRef: m.DefaultRef ?? "main",
-                Include: ImmutableArray.Create(m.Include ?? Array.Empty<string>()),
-                Exclude: ImmutableArray.Create(m.Exclude ?? Array.Empty<string>())))
+        return settings.Mappings
+            .Select(m => CreateMapping(settings.Defaults, m))
             .ToImmutableArray();
     }
 
+    private static SourceMapping CreateMapping(SourceMappingSetting defaults, SourceMappingSetting setting)
+    {
+        IEnumerable<string> include = setting.Include ?? Enumerable.Empty<string>();
+        IEnumerable<string> exclude = setting.Exclude ?? Enumerable.Empty<string>();
+
+        if (!setting.IgnoreDefaults)
+        {
+            if (defaults.Include is not null)
+            {
+                include = defaults.Include.Concat(include).ToArray();
+            }
+
+            if (defaults.Exclude is not null)
+            {
+                exclude = defaults.Exclude.Concat(exclude).ToArray();
+            }
+        }
+
+        return new SourceMapping(
+            Name: setting.Name ?? throw new InvalidOperationException($"Missing `name` in {SourceMappingsFileName}"),
+            Version: setting.Version,
+            DefaultRemote: setting.DefaultRemote ?? throw new InvalidOperationException($"Missing `defaultRemote` in {SourceMappingsFileName}"),
+            DefaultRef: setting.DefaultRef ?? defaults.DefaultRef ?? "main",
```

Comment:
Trivial comment - shouldn't the fallback `main` ref be configurable? Just in case we run into some specific repo

Reply:
You can configure the ref on the mapping itself in the JSON, then you can also configure a default ref for the whole `source-mapping.json` (example in the description) and only then when you don't do these two layers, it falls back to `main`.